classes/Cart.php

```diff
@@ -1943,11 +1945,45 @@ public function getOrderTotal(
             $type = Cart::BOTH_WITHOUT_SHIPPING;
         }
 
-        // filter products
+        // When cart is associated to order the computation may be different, especially for multi invoices orders
```

Comment:
I decided to modify the `Cart` class directly because too much of the logic was implemented inside the handler The problem being that some modules, and especially Payment modules use the `Cart::getOrderTotal` function to compute order amounts and compare them to validate their payments

So if the whole multi-invoice logic was absent from this legacy class, it wouldn't be able to compute correctly the values wouldn't match and therefore the payment would not be validated

Though it might seem scary to modify this method, it is supported by a few arguments:
- it only modifies the behaviour when cart has an associated Order, or an invoice id is provided (which is used internally by the handler)
- it doesn't change the behaviour of a regular cart in FO
- both cart computations and order computations are supported by more and more behat tests (which are all green)
- it is anyway necessary 😅